sunpy/database/database.py

```diff
@@ -446,6 +449,72 @@ def _download_and_collect_entries(self, query_result, **kwargs):
                 entry.download_time = datetime.utcnow()
                 yield entry
 
+    def _download_and_collect_fido_entries(self, search_result, **kwargs):
+
+        client = kwargs.pop('client', None)
+        path = kwargs.pop('path', None)
+        progress = kwargs.pop('progress', False)
+        methods = kwargs.pop('methods', ('URL-FILE_Rice', 'URL-FILE'))
+        overwrite = kwargs.pop('overwrite', False)
+
+        if kwargs:
+            k, v = kwargs.popitem()
+            raise TypeError('unexpected keyword argument {0!r}'.format(k))
+
+        entries_list = tables.entries_from_fido_search_result(search_result,
+            default_waveunit=self.default_waveunit)
+        entries_list = list(entries_list)
+
+        remove_list = []
+        delete_entries = []
+        for sr_entry, temp in zip(search_result, entries_list):
+            for database_entry in self:
+                if database_entry.path is not None and sr_entry._compare_attributes(database_entry,
+                    ["source", "provider", "physobs", "fileid", "observation_time_start",
+                    "observation_time_end", "instrument", "size", "wavemin", "wavemax"]):
+                    if not overwrite:
+                        remove_list.append(qr)
+                    else:
+                        delete_entries.append(database_entry)
+
+        for temp in remove_list:
+            search_result = [x for x in search_result if x != temp]
+
+        for temp in delete_entries:
+            self.remove(temp)
+
+        paths = Fido.fetch(search_result, progress=progress, path=path)
+
+        for (path, sr_entry) in zip(paths, entries_list):
+
+            try:
+                read_file_header(path)
```

Comment:
As discussed this probably shouldn't be here, as `entries_from_file` should raise if there is an error.

Reply:
My commit means you don't need this line anymore, I think